### PR TITLE
Use sortable set for "chunks" of Chunks

### PR DESCRIPTION
**Pull request by timse**
**What kind of change does this PR introduce?**

refactoring

**Did you add tests for your changes?**

yes

**If relevant, link to documentation update:**

N/A

**Summary**

Use SortableSet for "chunks" of chunks to increase performance. 
Hide them behind an API to simplify future reasoning about their usage.

**Does this PR introduce a breaking change?**

Maybe, probably not.

**Other information**

### CODE REVIEW

File: lib/Chunk.js [hunk 1]
```diff
@@ -61,6 +61,13 @@ class Chunk {
 		throw new Error("Chunk.initial was removed. Use isInitial()");
 	}
 
+	/**
+	 * @return {Set} - a set containing the chunks
+	 */
+	getChunks() {
+		return this._chunks;
```
Comment (timse) [inline lib/Chunk.js]: i assume you mean?
```
get chunksIterable() {
		return this._chunks[Symbol.iterator]();
}
```

is this needed anywhere?

Comment (timse) [inline lib/Chunk.js]: the other changes are implemented :)

Comment (sokra) [inline lib/Chunk.js]: > i assume you mean?

no I really meant the first one. returning an iterable, not an iterator.

Comment (sokra) [inline lib/Chunk.js]: it can be used in for of loops.

**Comment by sokra** [comment]
Thanks

**Pull request merged**